### PR TITLE
Add custom CSS, and extract repeated elements to _includes

### PR DESCRIPTION
**Pull request by tlambert03**
This PR cleans up a few things about styles and reduces a lot of duplicated code:

- adds `/assets/css/style.scss` ... to override a few things about the theme.  Particularly, it has a smaller max-width on the `.inner` class than it does on `img` tags, causing all of our images to overflow.  So this slightly increases the width of the text to line up with images (and also prevents horizontal scrolling on a lot of `<code>` blocks throughout the site.
- adds `/_includes/` folder where we can put things that get used a lot.  For instance, we have a very un-DRY `## all tutorials` at the bottom of every page... which will be a pain to maintain.  This puts that in `/_includes/footer.md` which we can include anywhere with 
```
{% include footer.md %}
```
- adds an `image.html` include to provide nicer images with captions when desired as follows:
```
{% include image.html url="./resources/image.gif" description="my caption" %}
```
- lastly, ignoring the `_sites` folder, which makes it easier to test this stuff locally (which I did using a local jekyll install) 

If anyone wants me to pull out any of these things into a separate PR, let me know

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,22 +33,4 @@ We're a community partner on the [imagesc forum](https://forum.image.sc/tags/nap
...
-- [surface layer tutorial](./tutorials/surface.md)
-- [vectors layer tutorial](./tutorials/vectors.md)
-- [gallery](./gallery/gallery.md)
-- [napari-tutorials on GitHub](https://github.com/napari/napari-tutorials)
-- [napari on GitHub](https://github.com/napari/napari)
-
-## applications
-
-- [using dask & dask_image to view and process large datasets with arbitrary formats](./tutorials/dask.md)
+{% include footer.md %}
```
Comment (sofroniewn) [inline README.md]: we clearly need this, but when I look at the readme markdown it doesn't render the actual footer which means that when people come to the readme of the tutorials repo the see this text here instead right? I don't love that, but I guess fewer people will be going to the repo than going to the website.

**Comment by tlambert03** [comment]
> when I look at the readme markdown it doesn't render the actual footer which means that when people come to the readme of the tutorials repo the see this text here instead right?

yeah, I agree... I hadn't thought of that.  The templating engine is run by jekyll... so won't work in the github readme.  We could potentially just let the readme have some duplicated code, just to keep those links active in the readme.  But generally, the more we reduce duplicate code and improve styles with `includes`, the less readable the markdown will be.  I would certainly be ok with keeping the main README.md readable on github though.  Lemme know and I can make that change.

> Also can you expand on how you're testing with "local jekyll install". I'm not so familiar with this stuff, if easy can you paste in here what you installed and the command you run? (or link out to some tutorial on this stuff I can check out)

Here's a good starting point...
[Testing github pages locally](https://help.github.com/en/github/working-with-github-pages/testing-your-github-pages-site-locally-with-jekyll)

but I definitely ran into a couple hiccups with the theme stuff.  You'll need to install an updated ruby (i used homebrew), along with [jekyll](https://jekyllrb.com/docs/installation/), and `gem install github-pages`... but I just kinda hacked my way through it.  if you have trouble, open an issue and I'll make a clearer step-by-step.  fwiw, here's what my `Gemfile` (toplevel of the repo) eventually looked like:
```
source "https://rubygems.org"
git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }
gem "jekyll", "= 3.8.5"
gem "github-pages", '202', group: :jekyll_plugins
```

**Comment by sofroniewn** [comment]
Ok thanks for the links, I'll look into that.

Maybe we add a section like `Website` into the readme that has a prominent link to `http://napari.org` so that if someone comes to the github page we redirect them to the website, could be at the bottom above the footer. If we do that then I'm ok having the stuff you have there now.

If you like that idea do you want to add it to this PR?

**Comment by tlambert03** [comment]
ok, right above the include, I added "napari-tutorials home page" ... i think that might be the best compromise.  I don't want it to say "go to website"... because it's also going to appear on the website.  so that seems like a decent middle ground.  It's pretty minor, so I'll merge this and we can certainly rearrange things in the future.

**Pull request merged**